### PR TITLE
Feat/dataset support api service

### PR DESCRIPTION
**Pull request by zxhlyh**

### CODE REVIEW

File: web/app/(commonLayout)/datasets/Container.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import ApiServer from './ApiServer'
+import Doc from './Doc'
+import TabSlider from '@/app/components/base/tab-slider'
+import { fetchDatasetApiBaseUrl } from '@/service/datasets'
+
+const Container = () => {
+  const options = [
+    {
+      value: 'dataset',
+      text: 'DATASETS',
```
Comment (crazywoola) [inline web/app/(commonLayout)/datasets/Container.tsx]: Seems missing i18n here

File: web/app/(commonLayout)/datasets/Container.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    },
+    {
+      value: 'api',
+      text: 'API ACCESS',
```
Comment (crazywoola) [inline web/app/(commonLayout)/datasets/Container.tsx]: Seems missing i18n here

File: web/app/(commonLayout)/datasets/template/template.en.mdx [hunk 3]
```diff
@@ -0,0 +1,791 @@
...
+    ```
+    </CodeGroup>
+  </Col>
+</Row>
+
+---
+
+<Heading
+  url='/datasets/{dataset_id}/documents/{document_id}/update_by_text'
+  method='POST'
```
Comment (crazywoola) [inline web/app/(commonLayout)/datasets/template/template.en.mdx:333]: Should we use PUT or PATCH for these kind of tasks.

File: web/app/(commonLayout)/datasets/template/template.en.mdx [hunk 4]
```diff
@@ -0,0 +1,791 @@
...
+            - segmentation (object) segmentation rules
+              - separator Custom segment identifier, currently only allows one delimiter to be set. Default is \n
+              - max_tokens Maximum length (token) defaults to 1000
+      </Property>
+    </Properties>
+  </Col>
+  <Col sticky>
+    <CodeGroup 
+      title="Request" 
+      tag="POST"
```
Comment (crazywoola) [inline web/app/(commonLayout)/datasets/template/template.en.mdx:378]: Should we use PUT or PATCH for these kind of tasks.

File: web/app/(commonLayout)/datasets/template/template.en.mdx [hunk 5]
```diff
@@ -0,0 +1,791 @@
...
+    }
+    ```
+    </CodeGroup>
+  </Col>
+</Row>
+
+---
+
+<Heading
+  url='/datasets/{dataset_id}/documents/{document_id}/update_by_file'
```
Comment (crazywoola) [inline web/app/(commonLayout)/datasets/template/template.en.mdx:429]: Should we use PUT or PATCH for these kind of tasks.

File: web/app/(commonLayout)/datasets/template/template.en.mdx [hunk 6]
```diff
@@ -0,0 +1,791 @@
...
+            - segmentation (object) segmentation rules
+              - separator Custom segment identifier, currently only allows one delimiter to be set. Default is \n
+              - max_tokens Maximum length (token) defaults to 1000
+      </Property>
+    </Properties>
+  </Col>
+  <Col sticky>
+    <CodeGroup 
+      title="Request" 
+      tag="POST"
```
Comment (crazywoola) [inline web/app/(commonLayout)/datasets/template/template.en.mdx:475]: Should we use PUT or PATCH for these kind of tasks.

File: web/app/(commonLayout)/datasets/template/template.zh.mdx [hunk 7]
```diff
@@ -0,0 +1,792 @@
...
+  </Col>
+</Row>
+
+---
+
+<Heading
+  url='/datasets/{dataset_id}/documents/{document_id}/update_by_text'
+  method='POST'
+  title='通过文本更新文档 '
+  name='#update_by_text'
```
Comment (crazywoola) [inline web/app/(commonLayout)/datasets/template/template.zh.mdx:336]: PUT or PATCH ?

File: web/app/(commonLayout)/datasets/template/template.zh.mdx [hunk 8]
```diff
@@ -0,0 +1,792 @@
...
+              - separator 自定义分段标识符，目前仅允许设置一个分隔符。默认为 \n
+              - max_tokens 最大长度 (token) 默认为 1000
+      </Property>
+    </Properties>
+  </Col>
+  <Col sticky>
+    <CodeGroup 
+      title="Request" 
+      tag="POST" 
+      label="/datasets/{dataset_id}/documents/{document_id}/update_by_text"
```
Comment (crazywoola) [inline web/app/(commonLayout)/datasets/template/template.zh.mdx:380]: PUT or PATCH ?

File: web/app/(commonLayout)/datasets/template/template.zh.mdx [hunk 9]
```diff
@@ -0,0 +1,792 @@
...
+    ```
+    </CodeGroup>
+  </Col>
+</Row>
+
+---
+
+<Heading
+  url='/datasets/{dataset_id}/documents/{document_id}/update_by_file'
+  method='POST'
```
Comment (crazywoola) [inline web/app/(commonLayout)/datasets/template/template.zh.mdx:431]: PUT or PATCH ?

File: web/app/(commonLayout)/datasets/template/template.zh.mdx [hunk 10]
```diff
@@ -0,0 +1,792 @@
...
+            - segmentation (object) 分段规则
+              - separator 自定义分段标识符，目前仅允许设置一个分隔符，默认为 \n
+              - max_tokens 最大长度 (token) 默认为 1000
+      </Property>
+    </Properties>
+  </Col>
+  <Col sticky>
+    <CodeGroup 
+      title="Request" 
+      tag="POST"
```
Comment (crazywoola) [inline web/app/(commonLayout)/datasets/template/template.zh.mdx:476]: PUT or PATCH?

**Review by crazywoola — CHANGES_REQUESTED** [review]

**Review by crazywoola — DISMISSED** [review]

**Review by iamjoel — DISMISSED** [review]

**Review by iamjoel — APPROVED** [review]

**Pull request merged**